File: src/main/java/raylras/zen/code/symbol/ParseTreeLocatable.java
package raylras.zen.code.symbol;

import org.antlr.v4.runtime.tree.ParseTree;
import raylras.zen.code.CompilationUnit;

import java.nio.file.Path;

public interface ParseTreeLocatable extends Locatable {

    ParseTree getCst();

    CompilationUnit getUnit();

    default Path getPath() {
        return getUnit().getPath();
    }

    default String getUri() {
        return getUnit().getPath().toUri().toString();
    }

}

File: core/src/main/kotlin/raylras/intellizen/symbol/ParseTreeLocatable.kt
package raylras.intellizen.symbol

import org.antlr.v4.runtime.tree.ParseTree
import raylras.intellizen.CompilationUnit
import java.nio.file.Path

interface ParseTreeLocatable : Locatable {
    val cst: ParseTree

    val unit: CompilationUnit

    override val path: Path
        get() = unit.path
}
